sdk/src/wallet/operations/transaction/high_level/delegation/create.rs

```diff
@@ -0,0 +1,201 @@
+                delegation_id: prepared.delegation_id,
+                transaction,
+            })
+    }
+
+    /// Prepares the transaction for [Wallet::create_delegation_output()].
+    pub async fn prepare_create_delegation_output(
+        &self,
+        params: CreateDelegationParams,
+        options: impl Into<Option<TransactionOptions>> + Send,
+    ) -> crate::wallet::Result<PreparedCreateDelegationTransaction> {
+        log::debug!("[TRANSACTION] prepare_create_delegation_output");
+        let protocol_parameters = self.client().get_protocol_parameters().await?;
+
+        let address = match params.address.as_ref() {
+            Some(bech32_address) => {
+                self.client().bech32_hrp_matches(bech32_address.hrp()).await?;
+                bech32_address.inner().clone()
+            }
+            None => self.address().await.inner().clone(),
+        };
+
+        let mut options = options.into();
+        let slot_commitment_id = if let Some(commitment_context) = options
+            .iter()
+            .flat_map(|opts| opts.context_inputs.iter())
+            .flat_map(|c| c.iter())
+            .find_map(|c| c.as_commitment_opt())
+        {
+            commitment_context.slot_commitment_id()
+        } else {
+            // Add a commitment context input with the latest slot commitment
+            let latest_id = self.client().get_info().await?.node_info.status.latest_commitment_id;
+            let context_input = ContextInput::from(CommitmentContextInput::new(latest_id));
+            if let Some(options) = &mut options {
+                if let Some(context_inputs) = &mut options.context_inputs {
+                    context_inputs.push(context_input);
+                } else {
+                    options.context_inputs.replace(vec![context_input]);
+                }
+            } else {
+                options.replace(TransactionOptions {
+                    context_inputs: Some(vec![context_input]),
+                    ..Default::default()
+                });
+            }
+            latest_id
+        };
+
+        let delegation_output_builder = DelegationOutputBuilder::new_with_amount(
+            params.delegated_amount,
+            DelegationId::null(),
+            params.validator_address,
+        )
+        .with_start_epoch(protocol_parameters.delegation_start_epoch(slot_commitment_id))
+        .add_unlock_condition(AddressUnlockCondition::new(address.clone()));
+
+        let output = delegation_output_builder.finish_output()?;
+
+        let transaction = self.prepare_transaction([output], options).await?;
+
+        Ok(PreparedCreateDelegationTransaction {
+            delegation_id: DelegationId::from(&transaction.transaction.id().into_output_id(0)),
+            transaction,
+        })
+    }
+
+    /// Gets an existing delegation output.
+    pub(crate) async fn get_delegation_output(&self, delegation_id: DelegationId) -> Option<OutputData> {
```

Comment:
We already have this, it's called `unspent_delegation_output`

Reply:
changed